app/domain/authentication/authn_k8s/inject_client_cert.rb

```diff
@@ -68,10 +69,14 @@ def validate
         @validate_pod_request.(pod_request: pod_request)
       end
 
+      def cert_installation_path
+        @cert_installation_path.nil? || @cert_installation_path.empty? ? DEFAULT_CERT_PATH : @cert_installation_path
+      end
```

Comment:
Are you not using memoization on purpose? i think that if you'd use memoization then in case it not nil but is empty it will not work because the logic will not run. If this is the case please add a comment on why we're not using memoization so a future developer will not change it.